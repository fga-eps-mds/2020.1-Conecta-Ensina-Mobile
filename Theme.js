import React from 'react';
import {ThemeProvider} from 'styled-components';

const theme = {
  colors: {
    fundoAzul: '#184E79',
    cinzaClaro: '#D9D9D9',
    cinzaEscuro: '#596C7C',
    vermelho: '#E46270',
    azulClaro: '#297FB8',
    verde: '#3FB93B',
    verdeEscuro: '#2D9D49',
    botÃ£oLoginNaoApertado: '#8B99A5',
    branco: '#ffffff',
    preto: 'black',
  },
  font: 'Roboto',
  fontSizes: {
    smaller: '12px',
    small: '14px',
    bigSmall: '16px',
    smallMedium: '20px',
    medium: '25px',
    bigMedium: '30px',
    big: '45px',
  },
};

const Theme = ({children}) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
