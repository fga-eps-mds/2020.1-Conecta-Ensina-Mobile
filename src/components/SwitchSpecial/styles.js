import styled from 'styled-components/native';

export const Container = styled.View`
  width: 280px;
  height: 45px;
  background-color: ${(props) => props.theme.colors.fundoAzul};
  justify-content: center;
  align-items: center;
  flex-direction: row;
  border-radius: 10px;
  margin-top: 5%;
`;

export const ContainerRow = styled.View`
  justify-content: center;
  align-items: center;
  flex: 1;
  padding-left: ${(props) => props.size.left};
  padding-right: ${(props) => props.size.right};
`;
