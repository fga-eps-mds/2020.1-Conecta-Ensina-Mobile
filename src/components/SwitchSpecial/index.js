import React from 'react';
import SwitchSelector from 'react-native-switch-selector';
import {Container, ContainerRow} from './styles';
import {withTheme} from 'styled-components/native';
import CustomText from '../CustomText';

function SwitchAdulthood({theme, onChange, onBlur, text}) {
  const estiloOptions = [
    {label: 'NÃ£o', value: false},
    {label: 'Sim', value: true},
  ];
  const size1 = {left: '10%', right: '5%'};
  const size2 = {left: '0%', right: '10%'};
  return (
    <Container>
      <ContainerRow size={size1}>
        <CustomText white smaller>
          {text}
        </CustomText>
      </ContainerRow>
      <ContainerRow size={size2}>
        <SwitchSelector
          options={estiloOptions}
          initial={0}
          onPress={(value) => onChange(value)}
          textColor={theme.colors.fundoAzul}
          selectedColor={theme.colors.preto}
          buttonColor={theme.colors.branco}
          borderColor={theme.colors.branco}
          backgroundColor={theme.colors.fundoAzul}
          borderRadius={5}
          buttonMargin={1}
          height={20}
          fontSize={12}
          onBlur={onBlur}
          style={{borderColor: 'white', borderWidth: 1, borderRadius: 5}}
        />
      </ContainerRow>
    </Container>
  );
}

export default withTheme(SwitchAdulthood);
