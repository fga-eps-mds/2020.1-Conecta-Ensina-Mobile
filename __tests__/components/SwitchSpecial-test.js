import React from 'react';
import 'react-native';
import {render, fireEvent} from '@testing-library/react-native';
import SwitchSpecial from '../../src/components/SwitchSpecial';
import Theme from '../../Theme';

describe('Testing SwitchSpecial component', () => {
  test('SwitchSpecial should be render', () => {
    const tree = render(
      <Theme>
        <SwitchSpecial />
      </Theme>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
