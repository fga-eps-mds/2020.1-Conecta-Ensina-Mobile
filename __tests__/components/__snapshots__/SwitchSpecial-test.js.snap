// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SwitchSpecial component SwitchSpecial should be render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#184E79",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginTop": "5%",
        "width": 280,
      },
    ]
  }
>
  <View
    size={
      Object {
        "left": "10%",
        "right": "5%",
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingLeft": "10%",
          "paddingRight": "5%",
        },
      ]
    }
  >
    <Text
      smaller={true}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Roboto",
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
      white={true}
    />
  </View>
  <View
    size={
      Object {
        "left": "0%",
        "right": "10%",
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingLeft": "0%",
          "paddingRight": "10%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "borderColor": "white",
            "borderRadius": 5,
            "borderWidth": 1,
          },
        ]
      }
    >
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "#184E79",
              "borderRadius": 5,
              "height": 22,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#ffffff",
                "borderRadius": 5,
                "borderWidth": 0,
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "black",
                      "fontSize": 12,
                      "fontWeight": "normal",
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                NÃ£o
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#184E79",
                      "fontSize": 12,
                      "fontWeight": "normal",
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                Sim
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
